tests/testsuite/package.rs

```diff
@@ -5149,23 +5149,105 @@ fn workspace_with_local_deps() {
             .file("level3/src/lib.rs", "")
             .build();
 
-    p.cargo("package")
+    p.cargo("package -Zpackage-workspace")
+        .masquerade_as_nightly_cargo(&["package-workspace"])
         .replace_crates_io(crates_io.index_url())
-        .with_status(101)
-        .with_stdout("")
-        .with_stderr(
+        .with_stdout_data("")
+        .with_stderr_data(
             "\
-[PACKAGING] level1 v0.0.1 ([CWD]/level1)
+[PACKAGING] level3 v0.0.1 ([ROOT]/foo/level3)
```

Comment:
You've handled the `with_stderr` deprecation but a large part of the intent behind that deprecation is to switch to snapshot testing, see #14039 for more details.

The switch in the assert method would ideally also be in the prior commit

Reply:
Ah, this snapshot testing is much nicer!